### DISCUSSION TITLE
UART logic levels - Teensy 4.1

### DISCUSSION THREAD

**Discussion by networker-x** [micropython/micropython#15770] [m.iMX RT / Teensy 4.x] [answered]
I want to use a Quecetel GPS&GSM module with the teensy 4.1. Now the GPS module use UART for communication and its logic level is 1.8V, now as far as I know the MIMXRT1062DVJ6B (the mcu in the teensy 4.1) supports 1.8V I/O, however in micropython how do I switch between 3.3v logic and 1.8v logic for this specific UART or is it autonegotiable?

**Comment by robert-hh** [comment, marked as answer]
That is not supported by the MIMXRT build. You may be able to reconfigure the hardware by direct memory writes to the registers using mem32() calls or viper code. Otherwise use external level shifters.
